### ISSUE TITLE
Where can a student see their grades?

### ISSUE THREAD

**Issue by maryamdev** [jupyter/nbgrader#987]
OS: Ubuntu 16.04.4 LTS
nbgrader version 0.5.4
jupyterhub version 0.9.0
jupyter notebook version 5.5.0

Expected behavior:
I want the student to be able to see his/her assignment grade. Is there a way for this?

Actual behavior:
As the Admin, after autograding an assignment, I can see the grade of each student. As a student, I can only see that an assignment has been submitted.

Thanks!

**Comment by jhamrick** [comment]
Students can run the notebook that you give them with the Validate extension which will tell them if all their tests passed, but they won't see the actual grade. After you've done the autograding, you can generate feedback (with the `nbgrader feedback` command---though unfortunately this isn't currently available in the formgrader interface) and return it to students which will show them their grades.

There's currently no support for showing students grades through the assignment list interface though it's something we could conceivably add in the future.

**Comment by maryamdev** [comment]
Hi Jessica! Many thanks!

This command generates an html file in the feedback folder:
`nbgrader feedback  "<assignment-name>"
`

However, it is generated in the instructor's folder:
/home/instructor/feedback/student

... instead of being in the student's folder:
/home/student

... which means by default the student will **not** see the feedback after generating it.

When I check the docs ([link](https://nbgrader.readthedocs.io/en/0.2.x/user_guide/06_returning_feedback.html)), it seems this is how it works.

So I am just double-checking with you (before writing my own scripts): is this the only current behaviour or do I miss something?

Many thanks!

**Comment by jhamrick** [comment]
Yes, that's the behavior currently---you will have to write your own scripts for returning feedback to students. We have plans to add the ability to return feedback to students through JupyterHub eventually (see discussion here: https://github.com/jupyter/nbgrader/issues/659 and https://github.com/jupyter/nbgrader/issues/436 ) but it's not clear what the timeline for this will be, unfortunately.

**Comment by maryamdev** [comment]
Thanks, Jessica!

**Comment by maryamdev** [comment]
For the record, I used [this script](https://gist.github.com/jhamrick/2ad63d7226b64c0a95fa/) written by Jessica to return the feedback to students' directories.

Because the feedback path is slightly different, I had only to change this line:
`feedback_dir = os.path.abspath(os.path.join("feedback", name))`


 ... to
`feedback_dir = os.path.abspath(os.path.join("feedback"))`

It works just fine. Thanks!

**Comment by perllaghu** [comment]
I read this to have two parts:

1. How to transfer feedback to the student
2. Let the student see a summary of their grade in the `Assignments` view

Number 1 has been available since 0.6.0

Number 2 is harder, as the value of the feedback is sitting in an element in a `.html` file - I see the following line:

```
        <h4>python_squares_assessment (demo) (Score: 1.5 / 8.0)</h4>
```
